examples/platforms/da15000/radio.c

```diff
@@ -0,0 +1,562 @@
+    hw_uart_write_buffer(HW_UART1, "\nR", 2);
+#endif
+    VerifyOrExit(s_state != kStateDisabled, error = kThreadError_Busy);
+    s_state = kStateReceive;
+    s_receive_frame.mChannel = aChannel;
+
+    uint32_t phyAckAttr;
+    phyAckAttr = 0x08 | ((aChannel - FTDF_OFFSET_CHANNEL) & 0xf) << 4 | (0 & 0x3) << 8;
+    FTDF_SET_FIELD(ON_OFF_REGMAP_PHYRXATTR, (((aChannel - FTDF_OFFSET_CHANNEL) & 0xf) << 4));
+    FTDF_SET_FIELD(ON_OFF_REGMAP_PHYACKATTR, phyAckAttr);
+
+    FTDF_SET_FIELD(ON_OFF_REGMAP_RXALWAYSON, 1);
+    FTDF_SET_FIELD(ON_OFF_REGMAP_RXENABLE, 1);
+
+exit:
+    return error;
+}
+
+RadioPacket *otPlatRadioGetTransmitBuffer(otInstance *aInstance)
+{
+    (void)aInstance;
+    return &s_transmit_frame;
+}
+
+ThreadError otPlatRadioTransmit(otInstance *aInstance, RadioPacket *aPacket)
+{
+    ThreadError error = kThreadError_None;
+    (void)aInstance;
+
+    FTDF_Boolean csmaSuppress = 0;
+    VerifyOrExit(s_state == kStateReceive, error = kThreadError_Busy);
+
+    ad_ftdf_send_frame_simple(aPacket->mLength, aPacket->mPsdu, aPacket->mChannel, 0, csmaSuppress); //Prio 0 for all.
+    s_state = kStateTransmit;
+    s_data_pending = false;
+#if DEBUG_LOG_ENABLE
+    hw_uart_write_buffer(HW_UART1, "\nT", 2);
+#endif
+
+exit:
+    return error;
+}
+
+
+int8_t otPlatRadioGetRssi(otInstance *aInstance)
+{
+    (void)aInstance;
+    return 0;
+}
+
+otRadioCaps otPlatRadioGetCaps(otInstance *aInstance)
+{
+    (void)aInstance;
+    return kRadioCapsNone;
+}
+
+void otPlatRadioSetPromiscuous(otInstance *aInstance, bool aEnable)
+{
+    (void)aInstance;
+    uint8_t value;
+    value = aEnable;
+    FTDF_setValue(FTDF_PIB_PROMISCUOUS_MODE, &value);
+    s_promiscuous = aEnable;
+}
+
+bool otPlatRadioGetPromiscuous(otInstance *aInstance)
+{
+    (void)aInstance;
+    return s_promiscuous;
+}
+
+ThreadError otPlatRadioHandleTransmitDone(bool *rxPending)
+{
+    ThreadError error = kThreadError_None;
+    VerifyOrExit(s_state == kStateTransmit, error = kThreadError_InvalidState);
+
+    s_state = kStateReceive;
+
+    if (rxPending != NULL)
+    {
+        *rxPending = s_data_pending;
+    }
+
+exit:
+    return error;
+}
+
+
+
+
+void FTDF_sendFrameTransparentConfirm(void *handle, FTDF_Bitmap32 status)
+{
+    (void)handle;
+    (void)status;
+    volatile uint32_t *txStatus = FTDF_GET_REG_ADDR_INDEXED(RETENTION_RAM_TX_RETURN_STATUS_1, FTDF_TX_DATA_BUFFER);
+
+    SendFrameDone = true;
+
+#if DEBUG_LOG_ENABLE
+    hw_uart_write_buffer(HW_UART1, "\nTD", 3);
+#endif
+}
+
+
+
+void da15000RadioProcess(otInstance *aInstance)
+{
+    bool rxPending;
+    uint8_t bufferForFTDFconfigValue = 0;
+
+    if (SendFrameDone)
+    {
+        FTDF_SET_FIELD(ON_OFF_REGMAP_RXENABLE, 1);
+        otPlatRadioHandleTransmitDone(&rxPending);
+        otPlatRadioTransmitDone(aInstance, &s_transmit_frame, false, kThreadError_None);
+        SendFrameDone = false;
+    }
+
+    if (SED)
+    {
+        if ((otPlatAlarmGetNow() - sleep_const_delay) > 100)
```

Comment:
Why is this two if statements, couldn't this just be a single if?  Is there any up side to having it as two?